integration-tests/tests/src/tests/sync/mongo-db-client.ts

```diff
@@ -0,0 +1,188 @@
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+import { ChangeEvent, Document, InsertEvent, MongoDBCollection } from "realm";
+
+import { authenticateUserBefore, importAppBefore } from "../../hooks";
+import { sleep } from "../../utils/sleep";
+
+type TestDocument = {
+  _id: number;
+  text?: string;
+  isLast?: boolean;
+};
+
+describe.skipIf(environment.missingServer, "MongoDB Client", function () {
+  this.timeout(60_000); // TODO: Temporarily hardcoded until envs are set up.
+  importAppBefore("with-db");
+  authenticateUserBefore();
+
+  const serviceName = "mongo-db";
+  const dbName = "test-database";
+  const collectionName = "test-collection";
+
+  describe("User", function () {
+    it("returns a MongoDB service when calling 'mongoClient()'", async function (this: AppContext & Mocha.Context) {
+      const service = this.app.currentUser?.mongoClient(serviceName);
+      expect(service).to.be.an("object");
+      expect(service).to.have.property("db");
+      expect(service).to.have.property("serviceName", serviceName);
+
+      const db = service?.db(dbName);
+      expect(db).to.be.an("object");
+      expect(db).to.have.property("name", dbName);
+      expect(db).to.have.property("collection");
+
+      const collection = db?.collection(collectionName);
+      expect(collection).to.be.an("object");
+      expect(collection).to.have.property("name", collectionName);
+      expect(collection).to.have.property("databaseName", dbName);
+      expect(collection).to.have.property("serviceName", serviceName);
+    });
+  });
+
+  describe("MongoDBCollection", function () {
+    let collection: MongoDBCollection<TestDocument>;
+
+    function getCollection<T extends Document = TestDocument>(currentUser: User | null): MongoDBCollection<T> {
+      if (!currentUser) {
+        throw new Error("A user must be authenticated before getting a MongoDB collection.");
+      }
+      return currentUser.mongoClient(serviceName).db(dbName).collection<T>(collectionName);
+    }
+
+    function deleteAllDocuments() {
+      return collection.deleteMany();
+    }
+
+    before(function (this: AppContext & Mocha.Context) {
+      collection = getCollection(this.app.currentUser);
+    });
+
+    beforeEach(async function () {
+      await deleteAllDocuments();
+    });
+
+    after(async function () {
+      await deleteAllDocuments();
+    });
+
+    describe("#watch", function () {
+      const text = "use some odd chars to force weird encoding %\n\r\n\\????>>>>";
```

Comment:
This looks similar to the upcoming tests from the [test migration](https://github.com/realm/realm-js/blob/test-migration-master/integration-tests/tests/src/tests/sync/user.ts#L426).
We will have to determine what is being covered by both.  In any case, I like that the mongo-client test are seperate from the `user` tests (I would never find them there.)

Reply:
The `"can watch changes correctly"` test in the test migration should test the same thing as in this file. I slightly refactored to make it more readable (dividing it into separate `it` tests rather than one big one).

Reply:
FWIW, I find these tests much more readable.  We placed a priority on functionality over readability in the test migration :-(   
So lets keep them both and replace later.